### PR TITLE
fix: support partial BaseWindow bounds

### LINKED ISSUE #52660: BaseWindow.setBounds throws an exception if some of the properties of the `Rectangle` are missing
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

43.1.0

### What operating system(s) are you using?

macOS

### Operating System Version

MacOS 27 (Golden Gate)

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

unknown

### Does the issue also appear in Chromium / Google Chrome?

No

### Expected Behavior

[Documentation](https://www.electronjs.org/docs/latest/api/base-window#winsetboundsbounds-animate) states that `BaseWindow::setBounds` accept `Partial<Rectangle>`.

This slightly modified example from docs should not throw any exceptions:

```ts
const { app, BaseWindow } = require('electron')

app.whenReady().then(() => {
  const win = new BaseWindow()

  // set all bounds properties
  win.setBounds({ x: 440, y: 225, width: 800, height: 600 })

  // set a single bounds property
  win.setBounds({ width: 100 })

  // { x: 440, y: 225, width: 100, height: 600 }
  console.log(win.getBounds())
})
```

### Actual Behavior

The provided example throws `UnhandledPromiseRejectionWarning: TypeError: Error processing argument at index 0`.

### Testcase Gist URL

_No response_

### Additional Information

Note, that it does not throw if I use `BrowserWindow` instead of `BaseWindow`.

### PR DESCRIPTION
**Pull request by DebadityaHait**
#### Description of Change

Moves partial-bounds normalization from `BrowserWindow` to `BaseWindow` so `BaseWindow.setBounds()` honors its documented `Partial<Rectangle>` contract. Current bounds are read only when a property is missing.

Fixes #52660

#### Checklist

- [x] I have filled out the PR description
- [x] I have reviewed and verified the changes
- [ ] `npm test` passes — requires a built Electron checkout; CI will run it
- [x] tests are changed or added
- [x] PR release notes are complete

#### Release Notes

Notes: none

### CODE REVIEW

File: lib/browser/api/base-window.ts [hunk 1]
```diff
@@ -11,6 +11,17 @@ BaseWindow.prototype._init = function (this: TLWT) {
   // Avoid recursive require.
   const { app } = require('electron');
 
+  const nativeSetBounds = this.setBounds;
+  this.setBounds = (bounds, ...opts) => {
+    if (!('x' in bounds && 'y' in bounds && 'width' in bounds && 'height' in bounds)) {
```
Comment (Imperat) [inline lib/browser/api/base-window.ts]: will it be better to compare against `undefined`?

`if (bounds.x == undefined || bounds.y == undefined || bounds.height == undefined || bounds.width == undefined)`

Consumers of the API can pass bounds like:

```
setBounds({ x: 0, y: 0, width: 500, height: undefined })
```

Comment (Imperat) [reply lib/browser/api/base-window.ts]: Thanks for adding this check. I think, it's better to have it rather than calling `this.getBounds` at all times

**Comment by MarshallOfSound** [comment]
Heads up: Windows CI builds moved to new runners on Monday and the old ones have now been removed. This branch predates that, so its Windows build jobs still point at the old runners and will sit queued forever if CI runs again. Rebasing on `main` (or "Update branch") picks up the change, nothing else needed.

**Comment by DebadityaHait** [comment]
Thanks! I updated the normalization to treat undefined properties as missing and added regression coverage for that case. I also rebased onto current `main` to pick up the Windows CI runner changes. JavaScript lint, formatting, and diff checks pass.

File: spec/api-browser-window-spec.ts [hunk 2]
```diff
@@ -89,6 +89,46 @@ describe('BrowserWindow module', () => {
...
+        const fullBounds = { x: 440, y: 225, width: 500, height: 400 };
+        const nativeGetBounds = baseWindow.getBounds.bind(baseWindow);
+        let getBoundsCalled = false;
+        baseWindow.getBounds = () => {
+          getBoundsCalled = true;
+          return nativeGetBounds();
+        };
+
+        baseWindow.setBounds(fullBounds);
+        expect(getBoundsCalled).to.be.false;
```
Comment (MarshallOfSound) [inline spec/api-browser-window-spec.ts]: `expect(...).to.be.false;` does not assert. The spec runner installs `dirty-chai`, so `true` and `false` are methods and must be called. All other boolean assertions in this file use the call form. As written, this test passes even if `getBounds()` is always called. Same on line 110.
```suggestion
        expect(getBoundsCalled).to.be.false();
```

File: spec/api-browser-window-spec.ts [hunk 3]
```diff
@@ -89,6 +89,46 @@ describe('BrowserWindow module', () => {
...
+
+        const boundsUpdate = { width: 200 };
+        baseWindow.setBounds(boundsUpdate as any);
+
+        expect(getBoundsCalled).to.be.true;
```
Comment (MarshallOfSound) [inline spec/api-browser-window-spec.ts]:
```suggestion
        expect(getBoundsCalled).to.be.true();
```

File: lib/browser/api/base-window.ts [hunk 4]
```diff
@@ -11,6 +11,20 @@ BaseWindow.prototype._init = function (this: TLWT) {
   // Avoid recursive require.
   const { app } = require('electron');
 
+  const nativeSetBounds = this.setBounds;
+  this.setBounds = (bounds, ...opts) => {
+    if (bounds.x === undefined || bounds.y === undefined || bounds.width === undefined || bounds.height === undefined) {
```
Comment (MarshallOfSound) [inline lib/browser/api/base-window.ts]: Nit: two out-of-contract inputs behave differently now. The old `BrowserWindow` wrapper spread `bounds`, so `win.setBounds(undefined)` was a no-op. This now throws `TypeError: Cannot read properties of undefined`. Also, the guard tests `=== undefined` but the fill uses `??`, so `{ x: null, width: 200 }` fills `x` from the current bounds while `{ x: null, y: 0, width: 200, height: 100 }` goes to native and throws. `bounds ?? {}` plus `== null` in the guard makes both paths agree.

**Review by MarshallOfSound — CHANGES_REQUESTED** [review]

**Comment by DebadityaHait** [comment]
Thanks! Addressed the requested edge cases:

- `bounds ?? {}` preserves `setBounds(undefined)` behavior.
- `== null` and `??` keep null/undefined handling consistent.
- Fixed the `dirty-chai` assertions to call `false()`/`true()`.
- Added regression coverage for undefined and null properties and an undefined bounds object.

I also rebased onto current `main`. JavaScript lint, formatting, and diff checks pass.

**Review by MarshallOfSound — APPROVED** [review]

**Comment by MarshallOfSound** [comment]
@DebadityaHait I think you need to rebase on latest main to pick up some new CI configurations

**Comment by DebadityaHait** [comment]
Rebased onto latest `main` in `ba9564b`; the patch itself is unchanged. Local lint, formatting, and diff checks pass. The refreshed CI run is currently waiting for workflow approval.

**Review by codebytere — APPROVED** [review]

**Review by VerteDinde — APPROVED** [review]
